### DISCUSSION TITLE
Uploading team info and your solutions

### DISCUSSION THREAD

**Discussion by fpetric** [larics/icuas22_competition#39] [Announcements]
# Uploading team info and your solutions

Dear Teams, 

we have sent emails to team leaders containing links to shared folders that we will use to process your team info and also your solutions. 

**Important**: If your team leader has not received a link, or you received a wrong link, let us know as soon as possible by sending email to competition email. The folder name should correspond to your team name, without spaces and special characters (such as `,`, `&` and similar).

## Uploading team info
Each folder contains a subfolder named `team_docs`. This folder is reserved for proofs of enrollment for team members, which should be scanned/photographed in color, in `pdf` format. You can submit individual documents for each team member, or a single pdf file for all members. Once you upload the documents, team leader should notify the organizers that the team registration has been finalized, using the official email of the competition. The email must contain:

- Team name (you can change the name if you want to)
- List of team members

## Uploading solutions
Your team folder will be used to submit your solutions. There are several options in which you can submit your code for evaluation, as listed below.

### Using Docker
You need to upload a `.zip` file named `<team_name>_docker.zip`. For team name, please use the name of the folder you were shared (team name without special characters). The zip file should contain:

- docker file that downloads and builds all the code
- all private ros packages that you developed that cannot be downloaded using the submitted docker file

### Without docker
If you do not want to use Docker, you need to upload a `.zip` file named `<team_name>_ws.zip`. For team name, please use the name of the folder you were shared (team name without special characters). The zip file should contain all developed and non-standard ROS packages from the `src` folder of your ROS workspace that are needed to build your code. You can assume we have the common ROS packages installed on the evaluation machine. Also, do not include `larics` packages that we provided (such as `uav_ros_simulation` or `larics_gazebo_worlds`) and their dependencies. 

### Instructions update April 2nd
As announced [here](https://github.com/larics/icuas22_competition/discussions/43#discussion-3982484), make sure to:

- include your version of `icuas22_competition` package (with your session.yml and any custom config files) to your team zip file.
- once you detect the tag, publish the position of the tag in world coordinate frame as a `geometry_msgs/Point` message to topic `/red/tag_position_reconstructed`
- along with the position of the tag, you should publish annotated image of the tag as a proof of detection (annotation in form of a bounding box around the tag is sufficient). The image should be published as a `sensor_msgs/Image` to topic `/red/tag_image_annotated`
- release the ball by publishing gain 0.0 on the uav topic `/red/uav_magnet/gain` and not on the ball magnet topic

### Important notes for solution submission

**If possible, please use Docker as that will probably mean that we will be able to run evaluations more often** 

**To increase the chance of your code working, please keep your submission as clean as possible. Never include build output from your machine in your submission.**

**Unlike the documentation, You do not need to inform us when you upload your solution by sending emails :)**

**More info on evaluation schedule will be released by April 1st. For a sneak peek of our plan, see #29**

**Comment by afaroo01** [comment]
Hi @fpetric,

Can you please answer this. How much time does it take to get the result of one evaluation? And do partial solution marks count towards the average of 5 best runs? I mean can we submit task 1 solution and get evaluated or it will be considered out of the 5 runs average. We are working towards tasks 2 and 3 but want to get our task 1 evaluated if doesn't affect the final scores.

**Reply by fpetric** [reply]
We evaluate the solution within a day of submission. With respect to 5 best runs, we will try to evaluate your final upload 5 times, so if those will be your best runs then any of the runs before will be discarded. As of now, we highly recommend submitting initial solution as soon as possible, we expect that there will be build/run issues and so on.

**Reply by afaroo01** [reply]
@fpetric 

Can you please elaborate on how to upload the solution using the docker way. We are new to using docker, hence are having difficulty following the steps:

-docker file that downloads and builds all the code
- all private ros packages that you developed that cannot be downloaded using the submitted docker file

Any resources or guide to follow to do these steps, please let know. Thanks

**Reply by fpetric** [reply]
if you are not comfortable with docker, you can also just submit a simple zip of your source folder from the workspace (don't include uav_ros_simulation, larics_gazebo_worlds and storm_gazebo_ros_magnet packages). Your zip should include your version of icuas22_competition package (with your session.yml and custom configs if you use them) and all your solution packages. If you are using some exotic package that is not standard, include that too. To summarize, the structure of your zip file should be:
```
<Team_name>.zip
|--- icuas22_competition
|--- <all_other_packages_used (except uav_ros_simulation, larics_gazebo_worlds, storgm_gazebo_ros_magnet)>
```

**Reply by afaroo01** [reply]
Hi @fpetric, we had sent you our latest solution yesterday. How much time will it require for its evaluation?

**Reply by fpetric** [reply]
See the latest update #65, evaluation results for current solutions will be sent on Monday. Sorry for a late reply.
